himbaechel/uarch/gowin/gowin.cc

```diff
@@ -110,20 +111,30 @@ struct GowinArch : HimbaechelArch
 
     bool match_device(const std::string &device) override { return device.size() > 2 && device.substr(0, 2) == "GW"; }
 
-    std::unique_ptr<HimbaechelAPI> create(const std::string &device,
-                                          const dict<std::string, std::string> &args) override
-    {
-        return std::make_unique<GowinImpl>();
-    }
+    std::unique_ptr<HimbaechelAPI> create(const std::string &device) override { return std::make_unique<GowinImpl>(); }
 } gowinArch;
 
+po::options_description GowinImpl::getUarchOptions()
+{
+    po::options_description specific("Gowin specific options");
+    specific.add_options()("family", po::value<std::string>(), "select Gowin family");
+    specific.add_options()("cst", po::value<std::string>(), "CST pin constraint file");
```

Comment:
"name of the constraints file". 
There are constraints not only for pins, but also for networks and primitives.